eZ/Publish/Core/Repository/SiteAccessAware/ContentService.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+/**
+ * ContentService class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\Repository\SiteAccessAware;
+
+use eZ\Publish\API\Repository\ContentService as ContentServiceInterface;
+use eZ\Publish\API\Repository\Values\Content\ContentCreateStruct;
+use eZ\Publish\API\Repository\Values\Content\ContentUpdateStruct;
+use eZ\Publish\API\Repository\Values\Content\ContentMetadataUpdateStruct;
+use eZ\Publish\API\Repository\Values\Content\LocationCreateStruct;
+use eZ\Publish\API\Repository\Values\ContentType\ContentType;
+use eZ\Publish\API\Repository\Values\User\User;
+use eZ\Publish\API\Repository\Values\Content\VersionInfo;
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\Core\Repository\Helper\LanguageResolver;
+
+/**
+ * SiteAccess aware implementation of ContentService injecting languages where needed.
+ */
+class ContentService implements ContentServiceInterface
+{
+    /** @var \eZ\Publish\API\Repository\ContentService */
+    protected $service;
+
+    /** @var LanguageResolver */
+    protected $languageResolver;
+
+    /**
+     * Construct service object from aggregated service and LanguageResolver.
+     *
+     * @param \eZ\Publish\API\Repository\ContentService $service
+     * @param LanguageResolver $languageResolver
+     */
+    public function __construct(
+        ContentServiceInterface $service,
+        LanguageResolver $languageResolver
+    ) {
+        $this->service = $service;
+        $this->languageResolver = $languageResolver;
+    }
+
+    public function loadContentInfo($contentId)
```

Comment:
Should we not use at least `@inheritdoc` all over this place?

Reply:
we can, any preference on that?

I think it was @Nattfarinn who mentioned this implies the exact same thing, and is more widely supported as `inheritdoc` is kind of only intended for the description only.

Reply:
Tbh, I don't have any preferences (just maybe typehinting everything over `@param` and `@return` ;) ) I just pointed it out just because as You said I thought we had discussion on that already.

Reply:
Right, so in this case where we don't have type hints, maybe we make a convention to use `inheritdoc` _(unsure if I can add type hint here, as i will then need to add it to API and then that will be a small break for anyone extending anything afaik. Also since id's are at least until now been defined as mixed)_

Reply:
Good for me :)

And yes, that would be a BC break, not even with just correcting typehinting but as I noticed Ids are somewhere passed as ints and somewhere(like in notifications) as string. So we would be surre that we have all cases tested for that. And of course for that specific case we would also decide if we should typehint those as ints (case of UUID?).

Reply:
> also decide if we should typehint those as ints (case of UUID?)

yes UUID is why they are documented as mixed so we have room to consider it for future storage engines.